### PR TITLE
WIP [dom] Eb dump

### PR DESCRIPTION
**Pull request by ekouts**
The files for PE19.09 were created by parsing and dumping the .eb files through EasyBuild, just to show how they would change. Many comments and local variables are eliminated, and the `toolchain = SYSTEM` is dumped as `toolchain = {'name': 'system', 'version': 'system'}`.

I have only done a dry-run, but the builds should be successful.

### CODE REVIEW

**Comment by gppezzi** [comment]
Thanks for this. It looks promising but the changes seem to be too disruptive: while I think we could live with loosing the current format, it seems that we're loosing comments 'randomly', so it would be better to see if we could avoid that. 

If we could have something that works if we only allow comments outside variable declarations, it could be a reasonable compromise.

File: easybuild/easyconfigs/b/bladeplugin/bladeplugin-0.1-CrayGNU-19.09.eb [hunk 1]
```diff
@@ -3,55 +3,31 @@ easyblock = 'ConfigureMake'
...
+description = "Simple OTF2 trace explorer"
+
 toolchain = {'name': 'CrayGNU', 'version': '19.09'}
 toolchainopts = {'verbose': False}
+
 source_urls = ['http://apps.fz-juelich.de/scalasca/releases/cube/4.4/dist']
 sources = [SOURCELOWER_TAR_GZ]
-checksums = ['305804fad578b1b6e4e188bd71b9b922']
 patches = ['%(name)s-%(version)s.patch']
+checksums = ['305804fad578b1b6e4e188bd71b9b922']  # ('papi/5.7.0.2', EXTERNAL_MODULE),
```
Comment (victorusu) [inline easybuild/easyconfigs/b/bladeplugin/bladeplugin-0.1-CrayGNU-19.09.eb]: The comment was moved here too

File: easybuild/easyconfigs/c/CDO/CDO-1.9.5-CrayGNU-19.09.eb [hunk 2]
```diff
@@ -15,24 +15,21 @@ toolchainopts = {'opt': True, 'pic': True}
 
 # Files visible at https://code.mpimet.mpg.de/projects/cdo/files,
 # however link is different for each version!
-source_urls = ['https://code.mpimet.mpg.de/attachments/download/18264/'] 
+source_urls = ['https://code.mpimet.mpg.de/attachments/download/18264/']
 sources = [SOURCELOWER_TAR_GZ]
-checksums = [ '0c60f2c94dc5c76421ecf363153a5043' ]
+checksums = ['0c60f2c94dc5c76421ecf363153a5043']  # No MPI support
```
Comment (victorusu) [inline easybuild/easyconfigs/c/CDO/CDO-1.9.5-CrayGNU-19.09.eb]: the comment was moved here too

File: easybuild/easyconfigs/c/CDO/CDO-1.9.5-CrayIntel-19.09.eb [hunk 3]
```diff
@@ -17,22 +17,19 @@ toolchainopts = {'opt': True, 'pic': True}
 # however link is different for each version!
 source_urls = ['https://code.mpimet.mpg.de/attachments/download/18264/']
 sources = [SOURCELOWER_TAR_GZ]
-checksums = [ '0c60f2c94dc5c76421ecf363153a5043' ]
+checksums = ['0c60f2c94dc5c76421ecf363153a5043']  # No MPI support
```
Comment (victorusu) [inline easybuild/easyconfigs/c/CDO/CDO-1.9.5-CrayIntel-19.09.eb]: the comment was moved here to

File: easybuild/easyconfigs/c/CrayPGI/CrayPGI-19.09.eb [hunk 4]
```diff
@@ -5,24 +5,21 @@ version = '19.09'
...
     ('atp/2.1.3', EXTERNAL_MODULE),
     ('pgi/19.4.0', EXTERNAL_MODULE),
     ('modules/3.2.11.3', EXTERNAL_MODULE),
-    # cray-libsci does not support PrgEnv-pgi, do not load it
-    # ('cray-libsci/19.06.1', EXTERNAL_MODULE),
     ('cray-mpich/7.7.10', EXTERNAL_MODULE),
     ('craype/2.6.1', EXTERNAL_MODULE),
     ('pmi/5.0.14', EXTERNAL_MODULE),
-]
+]  # ('cray-libsci/19.06.1', EXTERNAL_MODULE),
```
Comment (victorusu) [inline easybuild/easyconfigs/c/CrayPGI/CrayPGI-19.09.eb]: the comment was moved here too.

File: easybuild/easyconfigs/e/Extrae/Extrae-3.5.4-CrayGNU-19.09.eb [hunk 5]
```diff
@@ -17,45 +18,27 @@ description = """
 toolchain = {'name': 'CrayGNU', 'version': '19.09'}
 toolchainopts = {'verbose': False}
 
-source_urls = ['https://ftp.tools.bsc.es/extrae']
-sources = ['extrae-%(version)s-src.tar.bz2']
+source_urls = ['https://ftp.tools.bsc.es/%(namelower)s']
+sources = ['%(namelower)s-%(version)s-src.tar.bz2']  # ('Dimemas', '5.3.6', '', True),
```
Comment (victorusu) [inline easybuild/easyconfigs/e/Extrae/Extrae-3.5.4-CrayGNU-19.09.eb]: the comment was moved here too

File: easybuild/easyconfigs/e/Extrae/Extrae-3.5.4-CrayGNU-19.09.eb [hunk 6]
```diff
@@ -17,45 +18,27 @@ description = """
...
-configopts += ' --with-papi=$PAT_BUILD_PAPI_BASEDIR'
-configopts += ' --with-binary-type=64'
-configopts += ' --enable-parallel-merge'
-configopts += ' --enable-sampling'
-configopts += ' --with-clustering=$EBROOTCLUSTERINGSUITE'
-configopts += ' --with-spectral=$EBROOTSPECTRAL'
-configopts += ' --without-dyninst'
-# to build without cuda, use:
-configopts += ' --without-cuda'
+configopts = "  --with-libz=$EBROOTZLIB --with-unwind=$EBROOTLIBUNWIND --with-mpi=$CRAY_MPICH_DIR --enable-openmp --with-fft=$FFTW_DIR --with-papi=$PAT_BUILD_PAPI_BASEDIR --with-binary-type=64 --enable-parallel-merge --enable-sampling --with-clustering=$EBROOTCLUSTERINGSUITE --with-spectral=$EBROOTSPECTRAL --without-dyninst --without-cuda"
```
Comment (victorusu) [inline easybuild/easyconfigs/e/Extrae/Extrae-3.5.4-CrayGNU-19.09.eb]: this line is extremely long

File: easybuild/easyconfigs/e/Extrae/Extrae-3.7.1-CrayGNU-19.09.eb [hunk 7]
```diff
@@ -16,45 +17,27 @@ his or her application."""
...
-configopts += ' --with-papi=$PAT_BUILD_PAPI_LIBDIR/..'
-configopts += ' --with-binary-type=64'
-configopts += ' --enable-parallel-merge'
-configopts += ' --enable-sampling'
-configopts += ' --with-clustering=$EBROOTCLUSTERINGSUITE'
-configopts += ' --with-spectral=$EBROOTSPECTRAL'
-configopts += ' --without-dyninst'
-# to build without cuda, use:
-configopts += ' --without-cuda'
+configopts = "  --with-libz=$EBROOTZLIB --with-unwind=$EBROOTLIBUNWIND --with-mpi=$CRAY_MPICH_DIR --enable-openmp --with-fft=$FFTW_DIR --with-papi=$PAT_BUILD_PAPI_LIBDIR/.. --with-binary-type=64 --enable-parallel-merge --enable-sampling --with-clustering=$EBROOTCLUSTERINGSUITE --with-spectral=$EBROOTSPECTRAL --without-dyninst --without-cuda"
```
Comment (victorusu) [inline easybuild/easyconfigs/e/Extrae/Extrae-3.7.1-CrayGNU-19.09.eb]: this line is very long

File: easybuild/easyconfigs/g/googlebenchmark/googlebenchmark-1.5.0-CrayGNU-19.09.eb [hunk 8]
```diff
@@ -3,25 +3,26 @@ easyblock = 'CMakeMake'
...
+
 source_urls = ['https://github.com/google/benchmark/archive']
 sources = ['v%(version)s.tar.gz']
+
 builddependencies = [
     ('CMake', '3.14.5', '', True),
     ('FileCheck', '873043d'),
 ]
+
+preconfigopts = " pwd && cd ../benchmark-%(version)s && wget https://github.com/google/googletest/archive/release-1.8.1.tar.gz && tar xf release-1.8.1.tar.gz && ln -s googletest-release-1.8.1 googletest && cd - &&"
```
Comment (victorusu) [inline easybuild/easyconfigs/g/googlebenchmark/googlebenchmark-1.5.0-CrayGNU-19.09.eb]: this is a long line

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.09-APEX.eb [hunk 9]
```diff
@@ -16,37 +15,24 @@ the available resources to achieve unprecedented levels of scalability.
 toolchain = {'name': 'CrayGNU', 'version': '19.09'}
 toolchainopts = {'pic': True}
 
-builddependencies = [('CMake', '3.14.5', '', True)]
-
+source_urls = ['https://github.com/STEllAR-GROUP/%(namelower)s/archive']
+sources = ['%(version)s.tar.gz']
 patches = ['git_external.patch']
+checksums = ['cd34da674064c4cc4a331402edbd65c5a1f8058fb46003314ca18fa08423c5ad']  # papi provided by perftools-base
```
Comment (victorusu) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.09-APEX.eb]: the comment was moved here too

File: easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.09-APEX.eb [hunk 10]
```diff
@@ -16,37 +15,24 @@ the available resources to achieve unprecedented levels of scalability.
...
     ('Boost', '1.70.0', '-python3'),
     ('hwloc', '2.0.3', '', True),
     ('jemalloc', '5.1.0'),
     ('OTF2', '2.2', '', True),
-    # papi provided by perftools-base
 ]
 
-source_urls = ['https://github.com/STEllAR-GROUP/hpx/archive']
-sources = ['%(version)s.tar.gz']
+configopts = "-DHPX_WITH_MALLOC=JEMALLOC -DHPX_WITH_MAX_CPU_COUNT=128 -DHPX_WITH_PARCELPORT_MPI=ON -DHPX_WITH_PARCELPORT_MPI_MULTITHREADED=ON -DHPX_WITH_PAPI=ON -DHPX_WITH_APEX=ON -DHPX_WITH_APEX_TAG=v2.%(version_major_minor)s -DAPEX_WITH_PAPI=ON -DAPEX_WITH_OTF2=ON -DHWLOC_ROOT=$EBROOTHWLOC -DJEMALLOC_ROOT=$EBROOTJEMALLOC -DOTF2_ROOT=$EBROOTOTF2 -DPAPI_ROOT=$PAT_BUILD_PAPI_BASEDIR "
```
Comment (victorusu) [inline easybuild/easyconfigs/h/HPX/HPX-1.3.0-CrayGNU-19.09-APEX.eb]: this is an extremely long line

File: easybuild/easyconfigs/h/Horovod/Horovod-0.16.4-CrayGNU-19.09-tf-1.14.0.eb [hunk 11]
```diff
@@ -25,22 +15,17 @@ sources = [SOURCELOWER_TAR_GZ]
...
 
 dependencies = [
-    ('cray-python/%s' % pyver, EXTERNAL_MODULE),
-    ('TensorFlow', tfver, '-cuda-%s-python%s' % (cudaver, py_maj_ver)),
+    ('cray-python/3.6.5.7', EXTERNAL_MODULE),
+    ('TensorFlow', '1.14.0', '-cuda-10.1.168-python3'),
 ]
 
-skipsteps = ['build']
+preinstallopts = "module unload atp/2.1.3 && export HOROVOD_CUDA_HOME=$CUDATOOLKIT_HOME && export HOROVOD_GPU_ALLREDUCE=NCCL && export HOROVOD_MPICXX_SHOW='CC --cray-print-opts=all' && export HOROVOD_WITH_TENSORFLOW=1 && export HOROVOD_WITHOUT_PYTORCH=1 && "
```
Comment (victorusu) [inline easybuild/easyconfigs/h/Horovod/Horovod-0.16.4-CrayGNU-19.09-tf-1.14.0.eb]: this is very long line

File: easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayCCE-19.09.eb [hunk 12]
```diff
@@ -10,56 +10,39 @@ Computing"""
...
 
-source_urls = ['https://github.com/nerscadmin/IPM/archive']
+# verbose=False avoids the configure step to fail (-craype-verbose error)
+source_urls = ['https://github.com/nerscadmin/%(name)s/archive']
 sources = ['%(version)s.tar.gz']
-checksums = [ 'c01207ca942bef74c7d48df965fca99f' ]
 # Forcing $IPM_LOGDIR to archive all reports in a shared dir:
-patches = [ 'IPM-2.0.6-Cray-19.06.patch' ]
+patches = ['%(name)s-%(version)s-Cray-19.06.patch']
+checksums = ['c01207ca942bef74c7d48df965fca99f']  # under /apps/common/system/modulefiles/ploticus/'
```
Comment (victorusu) [inline easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayCCE-19.09.eb]: The comment has moved here too

File: easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayCCE-19.09.eb [hunk 13]
```diff
@@ -10,56 +10,39 @@ Computing"""
...
-preconfigopts += 'sed -i -e \'s*val1 count _count size-count_lo _ucount count_lo st_count*count_lo*\' configure && '
-preconfigopts += 'sed -i -e \'s-int argc, char \*-int argc, char **-\' configure && '
-buildopts = ' MPICC=cc MPIF77=ftn MPIFC=ftn FC=ftn '
-configopts  = ' --with-papi=$PAT_BUILD_PAPI_LIBDIR/..'
-# With Cray, we can use perftools' papi, otherwise:
-# configopts  = ' --with-papi=`pkg-config --variable=papi_prefix papi`'
-configopts += ' --with-libunwind=$EBROOTLIBUNWIND'
-configopts += ' CPPFLAGS="-I$EBROOTMXML/include "'
-configopts += ' LIBS="-L$EBROOTMXML/lib "'
+preconfigopts = "./bootstrap.sh && sed -i -e 's/-qversion/-qversion --version/g' configure && sed -i -e 's*val1 count _count size-count_lo _ucount count_lo st_count*count_lo*' configure && sed -i -e 's-int argc, char \*-int argc, char **-' configure && "
```
Comment (victorusu) [inline easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayCCE-19.09.eb]: this is very long line

File: easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayPGI-19.09.eb [hunk 14]
```diff
@@ -10,49 +10,35 @@ Computing"""
...
-configopts  = ' --with-papi=$PAT_BUILD_PAPI_LIBDIR/..'
-# With Cray, we can use perftools' papi, otherwise use:
-# configopts  = ' --with-papi=`pkg-config --variable=papi_prefix papi`'
-configopts += ' --with-libunwind=$EBROOTLIBUNWIND'
-configopts += ' CPPFLAGS="-I$EBROOTMXML/include "'
-configopts += ' LIBS="-L$EBROOTMXML/lib "'
-buildopts = ' MPICC=cc MPIF77=ftn MPIFC=ftn FC=ftn '
-
-postinstallcmds = ["mv bin/ipm_parse %(installdir)s/bin/ipm_parse.pl "]
+preconfigopts = """./bootstrap.sh && sed -i -e 's/-qversion/-qversion --version/g' configure && sed -ie "s-CXX=pgcpp-CXX=CC-g" configure && sed -ie "s-CC=pgcc-CXX=cc-g" configure && sed -ie "s-F77=pgf77-F77=ftn-g" configure && """
```
Comment (victorusu) [inline easybuild/easyconfigs/i/IPM/IPM-2.0.6-CrayPGI-19.09.eb]: this is an extremely long line

File: easybuild/easyconfigs/i/IPython/IPython-5.7.0-CrayGNU-19.09-python2.eb [hunk 15]
```diff
@@ -25,143 +15,35 @@ toolchain = {'name': 'CrayGNU', 'version': '19.09'}
...
+    ('ipython', '5.7.0', {'source_urls': ['https://pypi.python.org/packages/source/i/ipython/'], 'req_py_majver': '2', 'modulename': 'IPython', 'patches': ['ipython-5.7.0-jupyter.patch'], 'req_py_minver': '7'}),
 ]
-
-# specify that Bundle easyblock should run a full sanity check, rather than
-# just trying to load the module
-full_sanity_check = True
-
-modextrapaths = {'PYTHONPATH': ['lib/python%s/site-packages' % pyshortver]}
+modextrapaths = {'PYTHONPATH': ['lib/python2.7/site-packages']}
+postinstallcmds = ['mkdir %(installdir)s/startup && cp /apps/common/UES/easybuild/sources/i/%(name)s/00-detect-virtualenv-sitepackages.py %(installdir)s/startup && cp /apps/common/UES/easybuild/sources/i/%(name)s/py27launcher %(installdir)s']
```
Comment (victorusu) [inline easybuild/easyconfigs/i/IPython/IPython-5.7.0-CrayGNU-19.09-python2.eb]: this is a long line

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-22Aug2018-CrayGNU-19.09-cuda-10.1.eb [hunk 16]
```diff
@@ -3,48 +3,30 @@ easyblock = 'MakeCp'
...
 builddependencies = [
-    (('cudatoolkit/10.1.105_3.27-7.0.1.0_8.1__ga311ce7', EXTERNAL_MODULE)),
+    ('cudatoolkit/10.1.105_3.27-7.0.1.0_8.1__ga311ce7', EXTERNAL_MODULE),
     ('cray-fftw/3.3.8.3', EXTERNAL_MODULE),
-    ('cray-python/%s' %pyver, EXTERNAL_MODULE),
+    ('cray-python/2.7.15.7', EXTERNAL_MODULE),
 ]
 
-files_to_copy = [(['src/lmp*'], "bin")]
+prebuildopts = """ cd ./src &&  make yes-standard yes-user-cg-cmm yes-user-omp yes-user-reaxc yes-gpu yes-k-space yes-molecule yes-mpiio yes-rigid yes-misc yes-user-misc  &&  make no-latte no-voronoi no-reax no-poems no-meam no-kim no-kokkos no-mscg &&  make package-update &&  pushd ../lib/reax && make -f Makefile.gfortran && popd &&  pushd ../lib/gpu && sed -e "s/-march=bdver1//g" -e "s/sm_[0-9]*/sm_60/g" Makefile.xk7 > Makefile.gpu &&  make -f Makefile.gpu && popd &&  sed -e 's/mpicxx/CC -fopenmp/' ./MAKE/Makefile.mpi > ./MAKE/Makefile.omp &&  sed -i -e 's/mpicxx/CC/' ./MAKE/Makefile.mpi && """
```
Comment (victorusu) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-22Aug2018-CrayGNU-19.09-cuda-10.1.eb]: this is an extremely long line

File: easybuild/easyconfigs/l/LAMMPS/LAMMPS-22Aug2018-CrayGNU-19.09.eb [hunk 17]
```diff
@@ -3,42 +3,28 @@ easyblock = 'MakeCp'
...
+sources = ['stable_22Aug2018.tar.gz']
 
 builddependencies = [
     ('cray-fftw/3.3.8.3', EXTERNAL_MODULE),
-    ('cray-python/%s' %pyver, EXTERNAL_MODULE),
+    ('cray-python/2.7.15.7', EXTERNAL_MODULE),
 ]
 
-files_to_copy = [(['src/lmp*'], "bin")]
+prebuildopts = " cd ./src &&  make yes-standard yes-user-cg-cmm yes-user-omp yes-user-reaxc yes-user-misc &&  make no-latte no-voronoi no-reax no-poems no-meam no-kim no-kokkos no-mscg no-gpu &&  make package-update &&  pushd ../lib/reax && make -f Makefile.gfortran && popd &&  sed -e 's/mpicxx/CC -fopenmp/' ./MAKE/Makefile.mpi > ./MAKE/Makefile.omp &&  sed -i -e 's/mpicxx/CC/' ./MAKE/Makefile.mpi && "
```
Comment (victorusu) [inline easybuild/easyconfigs/l/LAMMPS/LAMMPS-22Aug2018-CrayGNU-19.09.eb]: this is an extremely long line

File: easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-CrayGNU-19.09.eb [hunk 18]
```diff
@@ -18,29 +18,24 @@ description = """The LLVM Core libraries provide a modern source- and target-ind
...
+]
 dependencies = [
     ('zlib', '1.2.11'),
-    ('cray-python/3.6.5.7', EXTERNAL_MODULE)
+    ('cray-python/3.6.5.7', EXTERNAL_MODULE),
 ]
 
-builddependencies = [('CMake', '3.14.1','', True)]
-
+configopts = "-DCMAKE_BUILD_TYPE:STRING=Release -DCMAKE_BUILD_WITH_INSTALL_RPATH:BOOL=ON -DBUILD_SHARED_LIBS=OFF -DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 -DLLVM_ENABLE_RTTI:BOOL=ON -DLLVM_BUILD_LLVM_DYLIB:BOOL=ON -DLLVM_ENABLE_ZLIB=ON"
```
Comment (victorusu) [inline easybuild/easyconfigs/l/LLVM/LLVM-5.0.0-CrayGNU-19.09.eb]: this is a very long line

File: easybuild/easyconfigs/l/LLVM/LLVM-8.0.0-CrayGNU-19.09.eb [hunk 19]
```diff
@@ -17,26 +16,23 @@ to use LLVM as an optimizer and code generator."""
...
 
 # required to install extra tools in bin/
-configopts  = " -DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86 "
-configopts += " -DLLVM_TARGETS_TO_BUILD=X86 "
-configopts += " -DLLVM_ENABLE_RTTI:BOOL=ON "
-configopts += " -DLLVM_BUILD_LLVM_DYLIB:BOOL=ON "
-configopts += " -DLLVM_ENABLE_ZLIB=ON "
-configopts += " -DCMAKE_BUILD_TYPE=Release "
-configopts += " -DBUILD_SHARED_LIBS=ON "
+configopts = " -DLLVM_INSTALL_UTILS=ON -DLLVM_TARGETS_TO_BUILD=X86  -DLLVM_TARGETS_TO_BUILD=X86  -DLLVM_ENABLE_RTTI:BOOL=ON  -DLLVM_BUILD_LLVM_DYLIB:BOOL=ON  -DLLVM_ENABLE_ZLIB=ON  -DCMAKE_BUILD_TYPE=Release  -DBUILD_SHARED_LIBS=ON "
```
Comment (victorusu) [inline easybuild/easyconfigs/l/LLVM/LLVM-8.0.0-CrayGNU-19.09.eb]: this is a very long line

File: easybuild/easyconfigs/l/Libint/Libint-1.1.4-CrayGNU-19.09.eb [hunk 20]
```diff
@@ -9,17 +9,15 @@ description = """Libint library is used to evaluate the traditional (electron re
...
 configopts = "--enable-deriv --enable-r12"
 
+
 sanity_check_paths = {
-    'files': ['include/lib%(x)s/lib%(x)s.h' % {'x': x} for x in ['deriv', 'int', 'r12']] +
-             ['include/libint/hrr_header.h', 'include/libint/vrr_header.h'] +
-             ['lib/lib%s.a' % x for x in ['deriv', 'int', 'r12']] +
-             ['lib/lib%s.so' % x for x in ['deriv', 'int', 'r12']],
-    'dirs':[],
+    'files': ['include/libderiv/libderiv.h', 'include/%(namelower)s/%(namelower)s.h', 'include/libr12/libr12.h', 'include/%(namelower)s/hrr_header.h', 'include/%(namelower)s/vrr_header.h', 'lib/libderiv.a', 'lib/%(namelower)s.a', 'lib/libr12.a', 'lib/libderiv.so', 'lib/%(namelower)s.so', 'lib/libr12.so'],
```
Comment (victorusu) [inline easybuild/easyconfigs/l/Libint/Libint-1.1.4-CrayGNU-19.09.eb]: this is a very long line

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-CrayGNU-19.09-python2.eb [hunk 21]
```diff
@@ -2,53 +2,33 @@ easyblock = 'CMakeMake'
...
-        'checksums': ['887b6433a30282cb002117da89b05812c770fd9469f93950ff3866ddd02bfc64'],
-        'buildcmd': 'build_ext',
-        'modulename': 'pygpu',
-    }),
+dependencies = [
+    ('cray-python/2.7.15.7', EXTERNAL_MODULE),
 ]
 
-modextrapaths = {'PYTHONPATH': 'lib/python%s/site-packages' % pyshortver}
+exts_list = [('libgpuarray', '0.7.6', {'modulename': 'pygpu', 'source_tmpl': '%(name)s-%(version)s.tar.gz', 'req_py_minver': '7', 'buildcmd': 'build_ext', 'source_urls': ['https://github.com/Theano/libgpuarray/releases/download/v%(version)s/'], 'checksums': ['887b6433a30282cb002117da89b05812c770fd9469f93950ff3866ddd02bfc64'], 'req_py_majver': '2'})]
```
Comment (victorusu) [inline easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-CrayGNU-19.09-python2.eb]: this is a very long line

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-CrayGNU-19.09-python3.eb [hunk 22]
```diff
@@ -2,53 +2,33 @@ easyblock = 'CMakeMake'
...
-        'checksums': ['887b6433a30282cb002117da89b05812c770fd9469f93950ff3866ddd02bfc64'],
-        'buildcmd': 'build_ext',
-        'modulename': 'pygpu',
-    }),
+dependencies = [
+    ('cray-python/3.6.5.7', EXTERNAL_MODULE),
 ]
 
-modextrapaths = {'PYTHONPATH': 'lib/python%s/site-packages' % pyshortver}
+exts_list = [('libgpuarray', '0.7.6', {'modulename': 'pygpu', 'source_tmpl': '%(name)s-%(version)s.tar.gz', 'req_py_minver': '6', 'buildcmd': 'build_ext', 'source_urls': ['https://github.com/Theano/libgpuarray/releases/download/v%(version)s/'], 'checksums': ['887b6433a30282cb002117da89b05812c770fd9469f93950ff3866ddd02bfc64'], 'req_py_majver': '3'})]
```
Comment (victorusu) [inline easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-CrayGNU-19.09-python3.eb]: this is a very long line

File: easybuild/easyconfigs/l/line_profiler/line_profiler-2.1-CrayGNU-19.09-python3.eb [hunk 23]
```diff
@@ -3,34 +3,27 @@ easyblock = 'PythonPackage'
...
 toolchain = {'name': 'CrayGNU', 'version': '19.09'}
 toolchainopts = {'verbose': False}
 
 source_urls = ['https://github.com/rkern/%(name)s/archive']
 sources = ['%(version)s.tar.gz']
 
-dependencies = [ ('PyExtensions', pyver) ] # Cython
+dependencies = [
+    ('PyExtensions', '3.6.5.7'),
+]  # Cython
```
Comment (victorusu) [inline easybuild/easyconfigs/l/line_profiler/line_profiler-2.1-CrayGNU-19.09-python3.eb]: the comment was moved here to

File: easybuild/easyconfigs/m/MUST/MUST-v1.6-rc1-CrayGNU-19.09.eb [hunk 24]
```diff
@@ -19,41 +20,37 @@ description = """
...
-installopts += " sed -i -e \"s@${EASYBUILD_BUILDPATH}/%(name)s/%(version)s/%(toolchain_name)s-%(toolchain_version)s/easybuild_obj@%(installdir)s@g\" %(installdir)s/bin/mustrun && "
-installopts += " sed -i -e \"s@${EASYBUILD_BUILDPATH}/%(name)s/%(version)s/%(toolchain_name)s-%(toolchain_version)s@%(installdir)s@g\" %(installdir)s/bin/mustrun && "
-installopts += " sed -i -e 's@externals/GTI/system-builder@@g' %(installdir)s/bin/mustrun && "
-#
-installopts += " ls -l %(installdir)s/bin/mustrun && echo xxx2 && set +x ; "
+installopts = """ && set -x && echo EASYBUILD_BUILDPATH=$EASYBUILD_BUILDPATH && ls -l %(installdir)s/bin/mustrun &&  sed -i -e "s@${EASYBUILD_BUILDPATH}/%(name)s/%(version)s/%(toolchain_name)s-%(toolchain_version)s/easybuild_obj@%(installdir)s@g" %(installdir)s/bin/mustrun &&  sed -i -e "s@${EASYBUILD_BUILDPATH}/%(name)s/%(version)s/%(toolchain_name)s-%(toolchain_version)s@%(installdir)s@g" %(installdir)s/bin/mustrun &&  sed -i -e 's@externals/GTI/system-builder@@g' %(installdir)s/bin/mustrun &&  ls -l %(installdir)s/bin/mustrun && echo xxx2 && set +x ; """
```
Comment (victorusu) [inline easybuild/easyconfigs/m/MUST/MUST-v1.6-rc1-CrayGNU-19.09.eb]: this is an extremely long line

**Pull request closed without merging**